examples/datasets/mnist8m.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import numpy as np
+from sklearn.datasets import load_svmlight_file
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import normalize
+from scipy.sparse import save_npz, load_npz
+
+class Mnist8m():
+
+    def __init__(self, cache_dir):
+        self.name = type(self).__name__
+        self.cache_dir = cache_dir
+        self.working_dir = os.path.join(self.cache_dir, self.name)
+        self.files = ['mnist8m.X_train.npz',
+                      'mnist8m.X_test.npz',
+                      'mnist8m.y_train.npy',
+                      'mnist8m.y_test.npy']
+
+    def __check_files_exist(self, files):
+        files_exist = True
+        for file in files:
+            files_exist &= os.path.isfile(os.path.join(self.working_dir, file))
+        return files_exist
+
+    def get_train_test_split(self):
+
+        if not self.__check_files_exist(self.files):
+
+            print("files do not exist")
+
+            p = subprocess.Popen(['mkdir', '-p', self.working_dir])
+            p.wait()
+
+            p = subprocess.Popen(['wget', 'https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass/mnist8m.scale.bz2'], cwd=self.working_dir)
+            p.wait()
+
+            p = subprocess.Popen(['bunzip2', 'mnist8m.scale.bz2'], cwd=self.working_dir)
+            p.wait()
+
+            X, y = load_svmlight_file(os.path.join(self.working_dir, "mnist8m.scale"))
+
+            X_train, X_test, y_train, y_test = train_test_split(
```

Comment:
Is mnist8m multiclass? does train_test_split preserve the class ratios? asking just to see if we can end up with imbalanced datasets.

Reply:
I've no idea, this is the same dataset that we used for the examples in the PowerAI distribution.

Reply:
Yes, it's multi-class, we used it for testing multi-class Logistic Regression. 
That's fine, the accuracy makes sense in its notebook example 

`https://github.com/IBM/snapml-examples/blob/tpa-start/examples/training/logistic_regression/example_mnist8m.ipynb`